webapp/src/css/targets.less

```diff
@@ -74,15 +89,18 @@
 
 .target {
   &.has-goal,
-  &.has-goal:not(.goal-met), .percent {
+  &.has-goal:not(.goal-met),
+  .percent {
+    background-color: @gray-ultra-light;
```

Comment:
I am very far from a CSS expert, so I may be on the wrong track here, but I think this addition here has caused a unintentional (or maybe it was on purpose?) change to the background-color of the target entries _on the aggregates page_! 

Here is what it looks like in `master` (note the rows for "My First CHW" and "My Second CHW" in the right-hand pane).  

![image](https://github.com/medic/cht-core/assets/8441903/59626453-7e00-4dd7-846a-335836916161)

In your branch, it looks like this (with the background color for those rows now being set to `gray-ultra-light`)

![image](https://github.com/medic/cht-core/assets/8441903/41772ad7-d786-40b9-88cb-409d7b56251d)

Reply:
Thanks for review @jkuester , the side effect in Target Aggregate was definitely not intentional. I will fix it.... but before I do, I have a question/blocker. 

I am curious how you ended up with filled entries for Targets and Target Aggregates ?  

I have come across the option of [uploading the default Test Data](https://docs.communityhealthtoolkit.org/apps/tutorials/local-setup/#upload-test-data). I have tried it, but when I run the command 
`cht --url=https://medic:password@localhost:5984 --accept-self-signed-certs csv-to-docs upload-docs`
 it fails perhaps due to how I should [connect to a local instance](https://docs.communityhealthtoolkit.org/apps/tutorials/local-setup/#note-on-connecting-to-a-local-cht-instance). 

Please help me figure out how to load the target aggregate data thanks.

Reply:
:+1: Honestly it has been a long time since I have tinkered with the `upload-docs` command. I hope that we did not inadvertently break something in the `default` test data...  :thinking: 

Typically, a lot of my manual testing is one-off cases that don't make to much sense to automate via CSV (or I want to use a script to spew in a ton of data and I use [this helper](https://github.com/jkuester/cht-data-helper)).  For manual testing of the `default` config, adding pregnancies is the key!  Those can trigger all kinds of tasks/targets/etc!

Now, when it comes to the Target Aggregates, I think you need two things: the proper config set for the target, and a user with the proper hierarchy.  

In your [targets config](https://github.com/medic/cht-core/blob/2a135a3850696cf8fae87c19993d77722ae70004/config/default/targets.js#L1), you will need to set the [`aggregate` property](https://docs.communityhealthtoolkit.org/apps/reference/targets/#targetsjs) to `true`.  Then use cht-conf to `compile-app-settings upload-app-settings`.

For the user, with the `default` config (which has the default place hierarchy) I like to create a "CHW Supervisor" user for a person associated with my top-level Health Facility (`district_hospital`).  Then, under that health facility, I will create one or more CHW Area's, each with a "CHW" person/user.  Then you can use those CHW users to create new households/patients and add some pregnancies.  Then, login as the CHW Supervisor and you should see all the aggregate targets!

(Let me know if some, or all, of this does not make sense! There is a lot of moving parts here and I am probably not explaining well....)

Reply:
Hi Josh, you are explaining well but here is what happens. 

1. I have added the `aggregate` property and set it to `true` in targets config. Unfortunately, when I run 
`cht --url=https://medic:password@localhost:5984 --accept-self-signed-certs compile-app-settings upload-app-settings`  
It takes like 3-4 minutes then it fails with 

```
INFO Checking that https://medic:****@localhost:5984/medic/ is available... 
(node:58691) Warning: Setting the NODE_TLS_REJECT_UNAUTHORIZED environment variable to '0' makes TLS connections and HTTPS requests insecure by disabling certificate verification.
(Use `node --trace-warnings ...` to show where the warning was created)
INFO Error: Failed to get a response from https://medic:****@localhost:5984/medic/. Maybe you entered the wrong URL, wrong port or the instance is not started. Please check and try again.
    at Object.available (/opt/homebrew/lib/node_modules/cht-conf/src/lib/api.js:113:13)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async module.exports (/opt/homebrew/lib/node_modules/cht-conf/src/lib/main.js:167:5)
    at async /opt/homebrew/lib/node_modules/cht-conf/src/bin/index.js:16:5 
 ```

2. I have a "CHW Supervisor user" and a CHW user. I have registered some pregnancies which I can see in the Targets tab, but unfortunately I cannot see the Target Aggregates because I have not been able to `upload app-settings ` cc @latin-panda

Reply:
Hi @jkuester  I have made some progress. The command now works, `compile-app-setting` works but `upload-app-settings` fails 

```
INFO compile-app-settings complete. 
INFO Starting action: upload-app-settings… 
INFO StatusCodeError: 404 - "{\"error\":\"not_found\",\"reason\":\"missing\"}\n"
    at new StatusCodeError (/opt/homebrew/lib/node_modules/cht-conf/node_modules/request-promise-core/lib/errors.js:32:15)
    at Request.plumbing.callback (/opt/homebrew/lib/node_modules/cht-conf/node_modules/request-promise-core/lib/plumbing.js:104:33)
    at Request.RP$callback [as _callback] (/opt/homebrew/lib/node_modules/cht-conf/node_modules/request-promise-core/lib/plumbing.js:46:31)
    at Request.self.callback (/opt/homebrew/lib/node_modules/cht-conf/node_modules/request/request.js:185:22)
    at Request.emit (node:events:513:28)
    at Request.<anonymous> (/opt/homebrew/lib/node_modules/cht-conf/node_modules/request/request.js:1154:10)
    at Request.emit (node:events:513:28)
    at IncomingMessage.<anonymous> (/opt/homebrew/lib/node_modules/cht-conf/node_modules/request/request.js:1076:12)
    at Object.onceWrapper (node:events:627:28)
    at IncomingMessage.emit (node:events:525:35)
    at endReadableNT (node:internal/streams/readable:1358:12)
    at processTicksAndRejections (node:internal/process/task_queues:83:21) 
ERROR 404 - "{\"error\":\"not_found\",\"reason\":\"missing\"}\n" 
```

Reply:
Connecting cht-conf to a local dev instance gets weird since it really wants to talk using HTTPS, but a typical local dev deployment is not setup to accept HTTPS connections.  This is where the `--local` parameter comes in.  It basically tells cht-conf to connect to the url defined in the `COUCH_URL` envar _and allow HTTP connections_.  

`COUCH_URL=http://medic:password@localhost:5984/medic cht --local compile-app-settings upload-app-settings` works fine for me locally. Note that my url in the envar is `http://...`.  (Of course, if you already have a value set in your terminal session for `COUCH_URL`, then you do not need to include that in the command.)

Reply:
That worked @jkuester .. Time to work on the requested changes.